### PR TITLE
replace atomicwrites with os.replace

### LINKED ISSUE #10114: Get rid of atomicwrites (unmaintained)
PyPI has started enforcing 2-factor-auth for maintainers of various popular packages: https://twitter.com/pypi/status/1545455297388584960

For context, here is the mail I got:

> Congratulations! A project you ('The_Compiler') maintain has been designated as a critical project on PyPI. You can see which project(s) has been designated at http://pypi.org/manage/projects/.
>
> As part of this effort, in the coming months maintainers of projects designated as critical, like yourself, will be required to enable two-factor authentication on their account in order to add new releases or otherwise modify a critical 
project.
>
> Since you already have two-factor authentication enabled on your account, there's nothing you need to do at this time.
>
> PS: To make it easier for maintainers like you to enable two-factor authentication, we're also distributing security keys to eligible maintainers. See http://pypi.org/security-key-giveaway/ for more details.

---

Unfortunately, this has caused the maintainer of `atomicwrites` to go on what I can only describe as a rampage, first deleting the project from PyPI (and then finding out it's not possible to restore it): https://github.com/untitaker/python-atomicwrites/issues/61

...to then simply declare the project as unmaintained outright: https://github.com/untitaker/python-atomicwrites/commit/d18328460520e18b4f197297f962d4444c5889b6

No matter what the outcome of this will be, IMHO, given those actions I do not feel comfortable with trusting this dependency for something as popular as pytest.

The library itself [is relatively simple](https://github.com/untitaker/python-atomicwrites/blob/master/atomicwrites/__init__.py), and we only use it on Windows. It's MIT-licensed. Should we just copy the parts we need into pytest instead?

### PR DESCRIPTION
**Pull request by asottile**
resolves #10114

diff is best viewed with [?w=1](https://github.com/pytest-dev/pytest/pull/10115/files?w=1)

### CODE REVIEW

**Review by nicoddemus — APPROVED** [review]
Thanks!

File: changelog/10114.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Replace `atomicwrites` dependency on windows with `os.replace`.
```
Comment (nicoddemus) [inline changelog/10114.improvement.rst:1]:
```suggestion
Replace `atomicwrites <https://github.com/untitaker/python-atomicwrites>`__ dependency on windows with `os.replace`.
```

Comment (nicoddemus) [reply changelog/10114.improvement.rst:1]: Also perhaps we can consider this `trivial` and backport it to 7.1.x?

Comment (asottile) [reply changelog/10114.improvement.rst:1]: yep will rename the file and amend when I'm at a computer

**Pull request merged**